code/script.js

```diff
@@ -230,96 +323,176 @@ const setSecret = () => {
+};
 
-// This function should be invoked when you click on 'Find Out' button.
+// Checks if the secret character has the chosen characteristic
 const checkQuestion = () => {
   const { category, value } = currentQuestion
+  console.log(currentQuestion)
+  let keep = false
+// checks if the values are the same but in a different way depending on whether the object value is a string or array 
+  if (typeof secret[category] === 'string'){
+    keep = (secret[category] === value)
+    }
 
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
-
-  } else if (category === 'accessories' || category === 'other') {
+  else if (Array.isArray(secret[category])){
+    keep = (secret[category].includes(value))
+  }
 
+  else {
+    console.log("a problem occured")
   }
+  alertUserOfAnswer(keep)
+  filterCharacters(keep)
+};
+
+const alertUserOfAnswer = (keep) => {
+  const { category, value } = currentQuestion
+  // Show the correct alert message for different categories
+    if (keep) {
+      if (category === 'sporeDispersal'){
+        alert(
+        `Yes, the mushroom has ${value}! Keep all the mushrooms with ${value}`
+      )}
+      else if (category === 'growsOnTrees') {
+        alert(
+          `Yes, the mushroom grows ${value}! Keep all the mushrooms that grow ${value}`
+        )}
+      else if (category === 'poisonousOrEdible') {
+        alert(
+          `Yes, the mushroom is ${value}! Keep all the mushrooms that are ${value}`
+        )}
+      else if (category === 'capOrBracketProperties'){
+        alert(
+          `Yes, the cap can be ${value}! Keep all the mushrooms with caps that can be ${value}`
+        )}
+      else if (category === 'stalkProperties'){
+        alert(
+          `Yes, the stalk can be/has ${value}! Keep all the mushrooms that can be/have ${value}`
+        )}
+    } else {
+      if (category === 'sporeDispersal'){
+        alert(`No, the mushroom doesn't have ${value}! Remove all mushrooms with ${value}`
+      )}
+      else if (category === 'growsOnTrees') {
+        alert(`No, the mushroom doesn't grow ${value}! Remove all mushrooms that grow ${value}`
+        )
+      }
+      else if (category === 'poisonousOrEdible') {
+        alert(`No, the mushroom isn't ${value}! Remove all mushrooms that are ${value}`
+        )
+      }
+      else if (category === 'capOrBracketProperties'){
+        alert(`No, the cap isn't ${value}! Remove all mushrooms that are ${value}`
+        )
+      }
+      else if (category === 'stalkProperties'){
+        alert(`No, the stalk doesn't have/isn't ${value}! Remove all mushrooms that have/are ${value}`
+        )
+      }
+    };
 }
 
 // It'll filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
```

Comment:
nice that you separate this part and the alertuser function that you have created. Becomes more easy to follow what is happening in the code!